api/client/proto/authservice.proto

```diff
@@ -1379,11 +1415,17 @@ service AuthService {
         returns (CreateRecoveryCodesWithTokenResponse);
 
     // GetMFAAuthenticateChallengeWithToken retrieves challenges for all mfa devices for the user
+    // currently signed in.
+    rpc GetMFAAuthenticateChallengeWithAuth(GetMFAAuthenticateChallengeWithAuthRequest)
+        returns (MFAAuthenticateChallenge);
+    // GetMFAAuthenticateChallengeWithToken retrieves challenges for all mfa devices for the user
     // defined in the token.
     rpc GetMFAAuthenticateChallengeWithToken(GetMFAAuthenticateChallengeWithTokenRequest)
         returns (MFAAuthenticateChallenge);
     // GetMFADevicesWithToken returns all mfa devices for the user defined in the token.
     rpc GetMFADevicesWithToken(GetMFADevicesWithTokenRequest) returns (GetMFADevicesResponse);
     // DeleteMFADeviceWithToken deletes a mfa device for the user defined in the token.
     rpc DeleteMFADeviceWithToken(DeleteMFADeviceWithTokenRequest) returns (google.protobuf.Empty);
+    // AddMFADeviceWithToken adds a mfa device for the user defined in the token.
+    rpc AddMFADeviceWithToken(AddMFADeviceWithTokenRequest) returns (google.protobuf.Empty);
```

Comment:
There is a lot of API surface for MFA already, but it seems we're effectively doubling it by adding new `WithToken` methods for the lot of it.

Have we considered adding this functionality on top of the existing RPCs? We could support privilege tokens as a way of authenticating in the streaming methods, cutting down the necessary interactions to the bare minimum. Let's talk about this before you go much further in the implementation, as what we are doing is going to impact each other's work.

Reply:
Resolving based on various conversations we had.